### PR TITLE
This test case test the user self register with the auth header passed

### PR DESCRIPTION
**Pull request by shavantha**
Purpose
This test case will call the scim2/Me endpoint to self register a user

Approach
This test case self register a user calling the scim2/Me endpoint and then asserts the user name. Next, the cleanup method will be called to delete the created user. The deletion is not through the scim2/Me endpoint but by passing the userId.

Tested environments
1.ubuntu 18.04
2.java version "1.8.0_161"
3.Identity Server with JDBC Userstore and mysql database

### CODE REVIEW

File: product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        setKeyStoreProperties();
+        client = HttpClients.createDefault();
+        super.init();
+    }
+
+    @Test(description = "1.1.2.1.2.15")
```
Comment (denuwanthi) [inline product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java]: Why is the description is "1.1.2.1.2.15"?
Is this some index?

File: product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    private  void selfRegister() throws  Exception {
+
+        scimUsersEndpoint = backendURL + SEPERATOR + Constants.SCIMEndpoints.SCIM2_ENDPOINT + SEPERATOR + Constants.SCIMEndpoints.SCIM_ANONYMOUS_USER;
```
Comment (denuwanthi) [inline product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java]: Can you please check wether this line exceeds 120 which is the standard size recommnded in WSO2 coding template

File: product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java (hunk 1 above)
Comment (shavantha) [reply product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java]: this is the number from which testgrid will identify the passed test case

File: product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java (hunk 2 above)
Comment (shavantha) [reply product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/AnonymousProvisioningTestCase.java]: thanks, fixed the line break issue

**Comment by shavantha** [comment]
closing this PR as a new PR https://github.com/wso2/product-is/pull/4154 is created

**Pull request closed without merging**